code/controllers/subsystem/job.dm

```diff
@@ -78,8 +78,8 @@ SUBSYSTEM_DEF(job)
 		SetupOccupations()
 	return type_occupations[jobtype]
 
-/datum/controller/subsystem/job/proc/AssignRole(mob/dead/new_player/player, rank, latejoin = FALSE)
-	JobDebug("Running AR, Player: [player], Rank: [rank], LJ: [latejoin]")
+// Checks for if player is Assignable to Role
+/datum/controller/subsystem/job/proc/CanAssignRole(mob/dead/new_player/player, rank)
```

Comment:
There isn't much reason to move the checks out of the proc into another one

Reply:
For the sake of re-usability and not duplicating the logic checks originally found in `/datum/controller/subsystem/job/proc/AssignRole`, those logic checks were moved to and recreated with `/datum/controller/subsystem/job/proc/CanAssignRole`

The same goes to `/datum/controller/subsystem/job/proc/GiveRandomJob`, where the grabbing of a random job has been moved and is now first done with the new proc `/datum/controller/subsystem/job/proc/GetRandomJob` where it finds a Job that's actually select-able by using the reusable proc `CanAssignRole()` to ensure that the a Role can be used by that player...otherwise find the next random Role (until it runs out and returns null)

Reply:
Please take a look at https://github.com/yogstation13/Yogstation/pull/7697/files/2ad5ab94e1fb9f7f48bc22e6e0f9c60e514d205d#diff-80ea298b5ebb3d34358d35f2393c6cec for my reasons of trying not to duplicate code